tests/TestSuite/Mutant/ResultTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+*
+* @category   Humbug
+* @package    Humbug
+* @copyright  Copyright (c) 2015 Pádraic Brady (http://blog.astrumfutura.com)
+* @license    https://github.com/padraic/humbug/blob/master/LICENSE New BSD License
+* @author     Thibaud Fabre
+*/
+namespace Humbug\Test\TestSuite\Mutant;
+
+use Humbug\TestSuite\Mutant\Result;
+
+class ResultTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetStatusCodeReturnsCorrectCode()
+    {
+        $this->assertEquals(Result::TIMEOUT, Result::getStatusCode(true, true, true));
+        $this->assertEquals(Result::TIMEOUT, Result::getStatusCode(false, false, true));
+        $this->assertEquals(Result::TIMEOUT, Result::getStatusCode(true, false, true));
```

Comment:
Hum... I know booleans are bad (I'm currently banging my head on the desk
with them), on the other hand, we need to map three boolean results into a
single scalar value, and ideally IMO, the decision should be encapsulated
to ensure consistent mapping. I *could* replace the boolean arguments with
the objects providing them, but it would only hide the issue.... not solve
it.
Le 9 mars 2015 22:54, "Rafał Wartalski" <notifications@github.com> a écrit :

> In tests/TestSuite/Mutant/ResultTest.php
> <https://github.com/padraic/humbug/pull/122#discussion_r26080606>:
>
> > +* @package    Humbug
> > +* @copyright  Copyright (c) 2015 Pádraic Brady (http://blog.astrumfutura.com)
> > +* @license    https://github.com/padraic/humbug/blob/master/LICENSE New BSD License
> > +* @author     Thibaud Fabre
> > +*/
> > +namespace Humbug\Test\TestSuite\Mutant;
> > +
> > +use Humbug\TestSuite\Mutant\Result;
> > +
> > +class ResultTest extends \PHPUnit_Framework_TestCase
> > +{
> > +    public function testGetStatusCodeReturnsCorrectCode()
> > +    {
> > +        $this->assertEquals(Result::TIMEOUT, Result::getStatusCode(true, true, true));
> > +        $this->assertEquals(Result::TIMEOUT, Result::getStatusCode(false, false, true));
> > +        $this->assertEquals(Result::TIMEOUT, Result::getStatusCode(true, false, true));
>
> For this PR it will be OK.
> Passing boolean two function always should burn red light. That means that
> you have two behaviors in your method. In fact you should have two methods.
> Here you have 3 booleans, so in fact you have 8 behaviors(8 separate
> methods).
> Each new boolean in this method multiplies original behaviors by 2. So
> complexity rises exponential.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/padraic/humbug/pull/122/files#r26080606>.
>